src/page/main-page/ui/video-block/index.tsx

```diff
@@ -0,0 +1,62 @@
+'use client'
+
+import React, { useEffect, useState } from 'react'
+import YouTube from 'react-youtube'
+
+import Wrap from '@/shared/ui/wrap'
+import Image from 'next/image'
+import img from '/public/main-page/video-block.jpg'
+import Icon from '@/shared/svg'
+
+import cl from './style.module.scss'
+import UseResizeVideo from '@/page/main-page/ui/video-block/lib/hooks'
+
+const VideoBlock = () => {
+  const [videoWatch, setVideoWatch] = useState<boolean>(false)
+  const [videoConfig, setVideoConfig] = useState({
+    height: '600',
+    width: '1065',
+    playerVars: {
+      autoplay: 1,
+    },
+  })
+
+  useEffect(() => {
+    if (videoWatch) {
+      document.body.style.overflow = 'hidden'
+    } else {
+      document.body.style.overflow = 'unset'
+    }
+  }, [videoWatch])
+
+  return (
+    <>
+      <Wrap>
+        <div className={cl.videoBlock}>
+          <h2>We are Createx Construction Bureau</h2>
+          <p>We are rightfully considered to be the best construction company in the USA.</p>
+          <div className={cl.img}>
+            <Image src={img} alt='video' width={1230} height={500} />
+            <div className={cl.play} onClick={() => setVideoWatch(true)}>
+              <Icon name='play' className={cl.play} />
+            </div>
+          </div>
+        </div>
+      </Wrap>
+      {videoWatch && (
+        <div
+          className={cl.videoWrap}
+          onClick={() => setVideoWatch(false)}
```

Comment:
if we are making a change to the state based on previous values it is better to use a callback